### PR TITLE
hypre: add v2.16.0

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -67,7 +68,9 @@ class Hypre(Package):
     conflicts("+shared@:2.12.99 platform=darwin")
 
     def url_for_version(self, version):
-        if version >= Version('2.12.0'):
+        if version >= Version('2.16.0'):
+            url = 'https://github.com/hypre-space/hypre/archive/v{0}.tar.gz'
+        elif version >= Version('2.12.0'):
```
[the remark this answers is not in the document]
Comment (rfalgout) [reply var/spack/repos/builtin/packages/hypre/package.py]: You can download everything from hypre-space as suggested.  All of the releases there are accurate and correct with respect to source code.  The SHA inconsistency is probably due to the fact that we used to build the PDFs of the manuals before pushing everything to LLNL/hypre (notice the extra 'docs' directory at LLNL/hypre).  Our previous development site was the LLNL institutional bitbucket repo, but we have now moved all of that to hypre-space/hypre.

Let me know if you have questions.  Thanks!

[the remark this answers is not in the document]
Comment (junghans) [reply var/spack/repos/builtin/packages/hypre/package.py]: All done.

**Comment by citibeth** [comment]
Can we also download 2.10 and 2.11 from the same website, thereby eliminating the need for `url_for_version()`?

**Comment by junghans** [comment]
Nope, `2.10.0b` is missing on https://github.com/hypre-space/hypre or at least it is named differently, so this always needs to be downloaded from the old website.

**Pull request merged**